src/v/kafka/client/client.cc

```diff
@@ -307,6 +307,11 @@ client::list_offsets(model::topic_partition tp) {
             {{.name{tp.topic},
               .partitions{
                 {{.partition_index{tp.partition}, .max_num_offsets = 1}}}}}}}});
+
+        auto ec = res.data.topics[0].partitions[0].error_code;
+        if (ec != error_code::none) {
+            co_return ss::coroutine::make_exception(partition_error(tp, ec));
```

Comment:
nit: there is some code that can be removed in call sites of list_offsets now that its behavior is to throw on errors in the response -- I did a quick search + `seq_writer::read_sync` looks like it's doing a now-redundant check of the response length + errors.

To do this cleanly (i.e. without ambiguity over who's meant to check errors) probably means changing this function to return the one topic's offsets rather than the whole response, so understandable if you don't want to bite it off in this PR.

Reply:
Done. A different error will propagate, but that's ok, because it can't happen.